src/lib.rs

```diff
@@ -98,104 +85,96 @@ pub enum Base {
 
 impl Base {
     /// Get the base code.
-    pub fn code(&self) -> &str {
+    pub fn code(&self) -> u8 {
         use Base::*;
 
         match *self {
-            Base1 => "1",
-            Base2 => "0",
-            Base8 => "7",
-            Base10 => "9",
-            Base16 => "f",
-            Base16Upper => "F",
-            Base32hex => "v",
-            Base32hexUpper => "V",
-            Base32hexpad => "t",
-            Base32hexpadUpper => "T",
-            Base32 => "b",
-            Base32Upper => "B",
-            Base32pad => "c",
-            Base32padUpper => "C",
-            Base32z => "h",
-            Base58flickr => "Z",
-            Base58btc => "z",
-            Base64 => "m",
-            Base64pad => "M",
-            Base64url => "u",
-            Base64urlpad => "U",
+            Base1 => b'1',
```

Comment:
Enum values have to be `isize`, so would have to write variants as `Base1 = b'1' as isize`, which is not terrible, but it's not great either.

From what I've tested and benched, an exhaustive match enum to value seems to be always optimized to a lookup table, so there isn't really any performance to be gained here. Plus you get the guarantee that if a new variant is added it must have a value assigned by the programmer.

Regarding `TryFrom` - I personally prefer to avoid dependencies when I can, especially for things that can be done with a simple `fn` returning a result.